packages/starrating/src/react/index.js

```diff
@@ -0,0 +1,179 @@
+      themeName
+    } = this.state
+    if (shouldFillFullStar) {
+      return this.makeIcon({ ...full, index })
+    } else if (shouldFillHalfStar) {
+      return themeName === themeDefaultName ? (
+        <DarkHalfStar key={index} />
+      ) : (
+        <LightHalfStar key={index} />
+      )
+    } else {
+      return this.makeIcon({ ...empty, index })
+    }
+  }
+
+  getHoverStateIcon(index, shouldFillFullStar) {
+    const {
+      themedIcons: { hover, empty }
+    } = this.state
+    const fill = shouldFillFullStar ? hover.fill : empty.fill
+    const id = shouldFillFullStar ? hover.id : empty.id
+    return this.makeIcon({ id, fill, index })
+  }
+
+  generateIcons() {
+    const { hoverIndex } = this.state
+    const { value } = this.props
+
+    const halfIntRoundedValue = Math.floor(value * 2) / 2
+
+    let fullStars = Math.floor(halfIntRoundedValue)
+    let halfStars = Math.floor(halfIntRoundedValue) !== halfIntRoundedValue
+
+    const isHovering = hoverIndex !== null
+
+    return new Array(TOTAL_STARS).fill(undefined).map((_, index) => {
+      let icon
+
+      if (isHovering) {
+        const shouldFillFullStar = index <= hoverIndex
+        icon = this.getHoverStateIcon(index, shouldFillFullStar)
+      } else {
+        const shouldFillFullStar = index < fullStars
+        const shouldFillHalfStar = index === fullStars && halfStars
+        icon = this.getNonHoverStateIcon(
+          index,
+          shouldFillFullStar,
+          shouldFillHalfStar
+        )
+      }
+      return icon
+    })
+  }
+
+  makeClickable(icons) {
+    const { onClick } = this.props
+
+    return icons.map((icon, index) => {
+      const ratingValue = index + 1
+      const ratingString = `Rate ${ratingValue} star${
+        ratingValue > 1 ? 's' : ''
+      }`
+      return (
+        <button
+          key={index}
+          tabIndex={-1}
+          style={css.button}
+          onClick={() => onClick(ratingValue)}
+          onMouseOver={() => this.setState({ hoverIndex: index })}
+          onMouseLeave={() => this.setState({ hoverIndex: null })}
+          aria-label={ratingString}
+        >
+          {icon}
+        </button>
+      )
+    })
+  }
+
+  render() {
+    const { onClick, value } = this.props
+
+    let StarIcons = this.generateIcons()
+    StarIcons = onClick ? this.makeClickable(StarIcons) : StarIcons
+    return (
+      <div>
+        {onClick && (
+          <label>
+            <span style={css.screenReaderContent}>Rate</span>
+            <input
+              style={css.screenReaderContent}
+              type="range"
+              onChange={event => onClick(event.target.value)}
+              min={1}
+              max={5}
+              step={1}
+            />
+          </label>
+        )}
+        <span style={css.screenReaderContent}>{`This is rated ${value}`}</span>
+        {StarIcons}
+      </div>
+    )
+  }
+}
+
+StarRating.propTypes = {
+  value: PropTypes.number,
+  onClick: PropTypes.func
```

Comment:
Feel like `onChange` might be more appropriate because it returning the new value and its change over time from the component.